### PR TITLE
java-microprofile: make the default context root respond nicely

### PR DESCRIPTION
**Pull request by barecode**
java-microprofile: make the default context root (localhost:9080) respond with something better than a 404

This change adds a simple Hello World resource to the MicroProfile starter and bind the application to the default context root so that it responds on localhost:9080. When accessed, the user will see "Hello from Appsody!"

Fixes: https://github.com/appsody/stacks/issues/30

### CODE REVIEW

**Comment by neeraj-laad** [comment]
@barecode  Welcome to Appsody! and thanks for your contribution.

We will review the PR shortly.

@Emily-Jiang would like to have your thoughts on this too.

File: incubator/java-microprofile/templates/default/src/main/java/dev/appsody/starter/HelloResource.java [hunk 1]
```diff
@@ -0,0 +1,17 @@
+package dev.appsody.starter;
```
Comment (Emily-Jiang) [inline incubator/java-microprofile/templates/default/src/main/java/dev/appsody/starter/HelloResource.java]: Graham Charters and I discussed this when we first agreed on what to add. We decided not to add this one as this file will be deleted first. We just added something absolutely needed.

File: index.yaml [hunk 2]
```diff
@@ -1,7 +1,7 @@
 apiVersion: v1
 generated: 2019-06-24T21:00:00+00:00
 projects:
-  java-microprofile:
+  java-microprofile-mct:
```
Comment (Emily-Jiang) [inline index.yaml]: Why do we need to add -mct?

**Comment by neeraj-laad** [comment]
@barecode Thanks for your contribution.

This looks like a good idea to keep all stacks consistent. Will get relevant people to review shortly.

File: incubator/java-microprofile/templates/default/src/main/java/dev/appsody/starter/HelloResource.java (hunk 1 above)
Comment (Emily-Jiang) [reply incubator/java-microprofile/templates/default/src/main/java/dev/appsody/starter/HelloResource.java]: I had a chat with Graham again this morning. According to our survey, developers don't want a lot of code to be generated as they have to delete. This file will be the first one to be deleted. I suggest not to add in. This stack is for production use. We need to keep it concise and useful. Therefore, I suggest not to add this file.

File: index.yaml (hunk 2 above)
Comment (neeraj-laad) [reply index.yaml]: The `mct` should be removed and version updated for the same stack.

**Comment by neeraj-laad** [comment]
@barecode Are you able to update this PR by:

- remove `mct` references
- update version numbers in stack.yaml

@Emily-Jiang One one hand I agree that we do not want to clutter the developer's workspace, but on the other hand we should provide some feedback / success message when hitting the default endpoint.

We are looking for a way to return something meaningful on the default endpoint to make user feel that they have the application up and running. Developers should be able to remove that single file or comment a couple of lines to remove this. 

- Node and Nodejs-express stacks use a simple plan text response - "Welcome from Appsody"
- Spring stack uses an index.html with a welcome message, some links to Spring and a list of URLs.

Let me know what your thoughts are.

**Comment by neeraj-laad** [comment]
This has been resolved in 0.2.2 version. Thanks for your contribution.

Please feel free to reopen this if you believe the issue has not been completely resolved.

**Pull request closed without merging**